docs/designs/rubygems-001.md

```diff
@@ -0,0 +1,547 @@
+# Initial Design for the Rubygems Package Manager
+
+- [Overview](#overview)
+- [Context](#context)
+- [Description](#description)
+    - [Ruby ecosystem overview](#i-ruby-ecosystem-overview)
+    - [Current implementation overview (Cachito)](#ii-overview-of-the-current-implementation-in-cachito)
+    - [Design for the Cachi2 implementation](#iii-design-for-the-implementation-in-cachi2)
+- [Decision](#decision)
+
+## Overview
+
+Design that covers the initial implementation of the Rubygem/Bundler package manager in Cachi2. It takes inspiration from the original implementation done in [Cachito](https://github.com/containerbuildsystem/cachito).
+
+| | |
+|---------|---------|
+|Author |[Michal Šoltis](msoltis@redhat.com) |
+|Co-author |[Bruno Pimentel](bpimente@redhat.com) |
+|Proposed on |August 01, 2024 |
+
+## Context
+In the effort to evolve Cachi2 to be a full solution for the prefetching feature of Cachito, we have decided to implement support to the currently missing package managers: Rubygems and Yarn v1. This design covers only the implementation of Rubygems, and the Yarn v1 design will follow.
+
+## Description
+
+### I. Ruby ecosystem overview
+
+#### Development prerequisites
+In order to execute the commands in the examples below, make sure you have the following packages installed in your
+environment:
+
+```bash
+sudo dnf install rubygems rubygems-bundler
+```
+
+Or use the official Ruby image from Docker hub:
+```bash
+podman run --rm -it docker.io/library/ruby:3.3.3 bash
+```
```

Comment:
This whole Ruby ecosystem overview (up to cachi2 impl.) is IMO the **actual** overview you want to have in this design doc instead of what it currently reads.